### PR TITLE
Bumping fosite version to use community maintained JWT implementation

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
The existing JWT implementation is no longer maintained and has a existing security CVE of HIGH. By increasing the fosite version we instead use the community maintained, backwards compatible JWT implementation which has no known security issues.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3368

## Follow-up issue
_NA_

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]

**Comment by kumare3** [comment]
cc @EngHabu / @wild-endeavor

**Review by eapolinario — APPROVED** [review]
